src/components/3d_editor/advanced_geometry/PythonTransformation.jsx

```diff
@@ -0,0 +1,352 @@
+                                    label="Selected Materials"
+                                    placeholder="Select materials"
+                                />
+                            )}
+                            renderTags={(value, getTagProps) =>
+                                value.map((option, index) => (
+                                    <Chip
+                                        label={`${index}: ${option.name}`}
+                                        // eslint-disable-next-line react/jsx-props-no-spreading
+                                        {...getTagProps({ index })}
+                                    />
+                                ))
+                            }
+                        />
+                    </Box>
+                    <Box
+                        sx={{
+                            display: "flex",
+                            flexDirection: "row",
+                            alignItems: "center",
+                            gap: 2,
+                            mt: 2,
+                        }}
+                    >
+                        <Autocomplete
+                            value={transformationsMap[transformationParameters.transformationKey]}
+                            getOptionLabel={(option) => option.name}
+                            options={Object.values(transformationsMap)}
+                            onChange={this.handleTransformationParametersChange}
+                            size="medium"
+                            sx={{ width: 600 }}
+                            renderInput={(params) => (
+                                <TextField
+                                    // eslint-disable-next-line react/jsx-props-no-spreading
+                                    {...params}
+                                    label="Transformation"
+                                    placeholder="Select transformation"
+                                />
+                            )}
+                        />
+                    </Box>
+
+                    <Box
+                        sx={{
+                            display: "flex",
+                            alignItems: "center",
+                            justifyContent: "flex-end",
+                            gap: 2,
+                            mt: 2,
+                            width: "100%",
+                        }}
+                    >
+                        <Typography variant="body1">{getStatusText()}</Typography>
+                        <Button
+                            variant="contained"
+                            color={isLoading ? "inherit" : "success"}
+                            onClick={this.handleRun}
+                        >
+                            Run
+                        </Button>
+                    </Box>
+                </Paper>
+            );
+        };
+
+        return (
+            <ThemeProvider theme={theme}>
```

Comment:
Why do we need another theme provider here? Should be only one for the whole MD.

Reply:
MD's theme is dark, and it causes issues with MUI styled components for some reason, to overcome that fast I've wrapped this Dialog in its own provider with light theme.